base/inference.jl

```diff
@@ -671,7 +679,8 @@ function getfield_tfunc(s00::ANY, name)
             if isa(sv, Module) && isa(nv, Symbol)
                 return abstract_eval_global(sv, nv)
             end
-            if (isa(sv, DataType) || isimmutable(sv)) && isdefined(sv, nv)
+            if (isa(sv, DataType) || isa(sv, SimpleVector) || isa(sv, TypeName)
```

Comment:
the `cache` and `linearcache` fields of TypeName are not constant

although, I guess neither is the `uid` field of DataType